contracts/USDCPool/USDCPool.sol

```diff
@@ -20,6 +20,14 @@ contract USDCPool is
     event Provide(address indexed account, uint256 amount);
     event WithdrawEvent(address indexed account, uint256 amount);
 
+    /***
+     * @notice   decimals       override the decimal function's
+     * */
+
+    function decimals() public view virtual override returns (uint8) {
+        return 6;
+    }
```

Comment:
Can you explain this function ? Why do we need this ?

Reply:
so if we do not override this function then the token created by USDCpool will be 18 decimal and we have to put the formula to convert that 18  to 6 decimals that are n**10uint(decimal())  while minting and burning the token and I think that is not a good practice and as of now we are only going to deposit USDC into sc then we can create our Wusdc with  6 decimal and we don't need to put that formula in our contract. and it will mint & burn according to the value of USDC/WUSDC passed by the user.